common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -93,9 +101,12 @@ public Collation(
       this.hashFunction = hashFunction;
       this.supportsBinaryEquality = supportsBinaryEquality;
       this.supportsBinaryOrdering = supportsBinaryOrdering;
+      this.supportsLowercaseEquality = collationName.equals("UTF8_BINARY_LCASE");
```

Comment:
Can you push `supportsLowercaseEquality` in constructor argument instead of string eq check here? You can make it default to false if needed, but I would rather be explicit in Collation construction.

Reply:
done